packages/site/cms-site/src/blocks/DamVideoBlock.tsx

```diff
@@ -1,30 +1,61 @@
 "use client";
 import * as React from "react";
-import styled from "styled-components";
+import styled, { css } from "styled-components";
 
 import { DamVideoBlockData } from "../blocks.generated";
 import { withPreview } from "../iframebridge/withPreview";
 import { PreviewSkeleton } from "../previewskeleton/PreviewSkeleton";
+import { VideoPreviewImage, VideoPreviewImageProps } from "./helpers/VideoPreviewImage";
 import { PropsWithData } from "./PropsWithData";
 
+interface DamVideoBlockProps extends PropsWithData<DamVideoBlockData> {
+    aspectRatio?: string;
+    sizes?: string;
+    renderPreviewImage?: (props: VideoPreviewImageProps) => React.ReactElement;
+}
+
 export const DamVideoBlock = withPreview(
-    ({ data: { damFile, autoplay, loop, showControls } }: PropsWithData<DamVideoBlockData>): JSX.Element => {
+    ({ data: { damFile, autoplay, loop, showControls, previewImage }, aspectRatio, sizes = "100vw", renderPreviewImage }: DamVideoBlockProps) => {
         if (damFile === undefined) {
             return <PreviewSkeleton type="media" hasContent={false} />;
         }
 
+        const [showPreviewImage, setShowPreviewImage] = React.useState(true);
+        const hasPreviewImage = previewImage && previewImage.damFile;
+
         return (
-            <Video autoPlay={autoplay} muted={autoplay} loop={loop} controls={showControls} playsInline>
-                <source src={damFile.fileUrl} type={damFile.mimetype} />
-            </Video>
+            <>
+                {hasPreviewImage && showPreviewImage ? (
+                    renderPreviewImage ? (
+                        renderPreviewImage({ onClick: () => setShowPreviewImage(false), image: previewImage, aspectRatio, sizes })
+                    ) : (
+                        <VideoPreviewImage onClick={() => setShowPreviewImage(false)} image={previewImage} aspectRatio={aspectRatio} sizes={sizes} />
+                    )
+                ) : (
+                    <Video
+                        autoPlay={autoplay || (hasPreviewImage && !showPreviewImage)}
```

Comment:
I don't think this is correct. Wouldn't this start autoplay too soon for a video that has autoplay enabled and a preview image?

Reply:
this is ok, the video is not rendered behind the previewimage